rhel/vex/parser.go

```diff
@@ -183,31 +183,46 @@ type creator struct {
 	rc                 *repoCache
 }
 
-// WalkRelationships attempts to resolve a relationship until we have a package product_id and
-// a repo product_id. Relationships can be nested. If the pkgID and the repoID are the same we
-// either found no relationship or a relationship where both ends are pointing to the same
-// product_id, either way we don't have enough data to create a vulnerability.
-func walkRelationships(productID string, doc *csaf.CSAF) (string, string, error) {
-	pkgID, repoID := extractProductNames(productID, productID, doc)
-	if pkgID == repoID {
-		return "", "", fmt.Errorf("could not extract a distict pkgID and repoID from %q", productID)
+// WalkRelationships attempts to resolve a relationship until we have a package product_id,
+// a repo product_id and possibly a package module product_id. Relationships can be nested.
+// If we don't get an initial relationship or we don't get two component parts we cannot
+// create a vulnerability. We never see more than 3 components in the wild but if we did
+// we'd assume the component next to the repo product_id is the package module.
```

Comment:
do we have any logs output when unexpected formatting happens (like a 4th components)?

Reply:
>2 components (i.e. more complex relationships) are valid as per the spec so we account for it in the return values, so there is no warning currently.